# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../flex/flex-2.5.35-gcc44-1.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Submitted by: Matt Burgess (matthew at linuxfromscratch dot org)
Date: 2009-05-03
Initial Package Version: 2.5.35
Origin: Matt Burgess
Upstream Status: Submitted (attached to sourceforge bug 2178663)
Description: Fixes an error caused by header cleanups in GCC 4.4.0 that is
             evident from the test suite and would affect any C++ lexers
             generated by Flex.  Without this patch, Flex will generate lexers
             containing references to the 'EOF' symbol without including the
             necessary C++ header file, leading to:

             error: 'EOF' was not declared in this scope 

diff -Naur flex-2.5.35.orig/skel.c flex-2.5.35/skel.c
--- flex-2.5.35.orig/skel.c	2008-02-26 21:34:19.000000000 +0000
+++ flex-2.5.35/skel.c	2009-05-03 15:18:14.000000000 +0000
@@ -284,6 +284,7 @@
   "/* begin standard C++ headers. */",
   "#include <iostream> ",
   "#include <errno.h>",
+  "#include <cstdio>",
   "#include <cstdlib>",
   "#include <cstring>",
   "/* end standard C++ headers. */",
